kitsune/search/v2/base.py

```diff
@@ -306,27 +310,46 @@ class SumoSearch(SumoSearchInterface):
     class inherits, relevant to the documents the child class is searching over.
     """
 
-    results_per_page: int = settings.SEARCH_RESULTS_PER_PAGE
-    total: int = 0
-    hits: list[AttrDict] = dfield(default_factory=list)
-    results: list[dict] = dfield(default_factory=list)
+    total: int = dfield(default=0, init=False)
+    hits: list[AttrDict] = dfield(default_factory=list, init=False)
+    results: list[dict] = dfield(default_factory=list, init=False)
+    last_key: Union[int, slice, None] = dfield(default=None, init=False)
+
+    query: str = ""
+    default_operator: str = "AND"
+
+    def __len__(self):
+        return self.total
+
+    @overload
+    def __getitem__(self, key: int) -> dict:
+        ...
+
+    @overload
+    def __getitem__(self, key: slice) -> list[dict]:
+        ...
+
+    def __getitem__(self, key):
```

Comment:
This is clever but it feels too hacky at the same time. You pass  a Compound search object to the paginator and when it tries to get a page you execute the query here and then it returns the result list. The paginator should get the results in the first place and not the query. It works now but it also couples unrelated things together. Moreover, `__getitem__` should not be the method that executes the actual query.